packages/engine/tests/simulations/experiment.rs

```diff
@@ -0,0 +1,269 @@
+                serde_json::to_string_pretty(b).unwrap(),
+            )
+        }
+        (Value::Object(a), Value::Object(b)) => a.into_iter().for_each(|(key, sub_value)| {
+            if let Some(super_value) = b.get(key) {
+                assert_subset_object(sub_value, super_value, format!("{path}.{key}"));
+            } else {
+                panic!(
+                    "{path:?}: {key:?} is not present output.\nexpected: {}\ngot: {}",
+                    serde_json::to_string_pretty(a).unwrap(),
+                    serde_json::to_string_pretty(b).unwrap(),
+                )
+            }
+        }),
+        _ => {
+            panic!(
+                "{path:?}: Expected `{} == {}`",
+                serde_json::to_string_pretty(subset).unwrap(),
+                serde_json::to_string_pretty(superset).unwrap()
+            );
+        }
+    }
+}
+
+impl ExperimentOutput {
+    pub fn assert_subset_of(&self, superset: &Self) {
+        if let Some(sub_value) = &self.json_state {
+            assert_subset_object(
+                sub_value,
+                superset.json_state.as_ref().unwrap(),
+                String::from("json_state.json: "),
+            )
+        }
+        if let Some(sub_value) = &self.globals {
+            assert_subset_object(
+                sub_value,
+                superset.globals.as_ref().unwrap(),
+                String::from("globals.json: "),
+            )
+        }
+        if let Some(sub_value) = &self.analysis_outputs {
+            assert_subset_object(
+                sub_value,
+                superset.analysis_outputs.as_ref().unwrap(),
+                String::from("analysis_outputs.json: "),
+            )
+        }
+    }
+}
+
+pub fn read_config(path: impl AsRef<Path>) -> Result<Vec<(ExperimentType, Vec<ExperimentOutput>)>> {
+    #[derive(Deserialize)]
+    #[serde(rename_all = "kebab-case")]
+    struct Simple {
+        experiment_name: String,
+        expected_outputs: Vec<ExperimentOutput>,
+    }
+
+    #[derive(Deserialize)]
+    #[serde(rename_all = "kebab-case")]
+    struct SingleRun {
+        num_runs: u64,
+        expected_output: ExperimentOutput,
+    }
+
+    #[derive(Deserialize)]
+    struct Config {
+        #[serde(default)]
+        experiments: Vec<Simple>,
+        #[serde(default)]
+        runs: Vec<SingleRun>,
+    }
+
+    let config: Config = parse_file(path).wrap_err("Could not read configuration")?;
+    let simple_iter = config.experiments.into_iter().map(
+        |Simple {
+             experiment_name,
+             expected_outputs,
+         }| { (ExperimentType::Simple { experiment_name }, expected_outputs) },
+    );
+    let single_run_iter = config.runs.into_iter().map(
+        |SingleRun {
+             num_runs,
+             expected_output,
+         }| {
+            (ExperimentType::SingleRun { num_runs }, vec![
+                expected_output,
+            ])
+        },
+    );
```

Comment:
I find the formatting of both of these pretty unfriendly. Think we could play around with it a little?